app/controllers/annotations_controller.rb

```diff
@@ -154,6 +154,14 @@ def update
         return
       end
     end
-    @annotation_text.update(content: params[:content])
+
+    if params[:change_one]
+      new_text = @annotation_text.dup
+      new_text.update(content: params[:content], annotation_category_id: nil)
+      new_text.save
```

Comment:
`save` after `update` is redundant. Also let's wrap the two updates in a [transaction](https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html).